File: two-pointers/ValidPalindrome/validPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String newStr = String.join("", s).toLowerCase();
        int j = newStr.length() - 1;
        for (int i = 0; i < newStr.length(); i++) {
            if (Character.isLetterOrDigit(s.charAt(i)) && Character.isLetterOrDigit(s.charAt(j)))
                if (s.charAt(i) != s.charAt(j))
                    return false;
            j--;
        }
        return true;
    }
}

File: two-pointers/ValidPalindrome/validPalindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        newStr = ''.join(c for c in s if c.isalnum())
        j = len(newStr) - 1
        for i in range(len(newStr)):
            if newStr[i].lower() != newStr[j].lower():
                return False
            j -= 1

        return True